reactor-core/src/main/java/reactor/core/publisher/FluxOperator.java

```diff
@@ -46,6 +45,27 @@ protected FluxOperator(Flux<? extends I> source) {
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
+	public void subscribe(CoreSubscriber<? super O> subscriber) {
```

Comment:
@smaldini should we immediately bite the bullet and make this a breaking binary change? making this `final` here would force any implementor of `FluxOperator`/`MonoOperator` to use the subscribeOrReturn model (but then it becomes part of the public API)

Reply:
not doing this in 3.3.0 timeframe would push that optimization back to 3.4 for third operators. We can certainly do that internally for now, which would allow us to easily back track on this change during 3.3.x, though we'd need to keep in mind that any feature built on top of the tail call elimination would then only be impacting the core operators and not third-party ones, so we couldn't fully rely on them.

Reply:
I would not make it a breaking change, at least not in 3.3.x where we introduce the optimization. If we break it, there will be no version of a third-party operator that works with both 3.2 and 3.3, and we need one for the migration period

Reply:
And we do have major users of this API:
https://github.com/alibaba/Sentinel/blob/fc550343c767fdbf11cd17018c47bcc4513b9ae0/sentinel-adapter/sentinel-reactor-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/reactor/MonoSentinelOperator.java#L28

https://github.com/resilience4j/resilience4j/blob/96c7b53134d4155be7e07ab8065ffef08cd46a1d/resilience4j-reactor/src/main/java/io/github/resilience4j/reactor/circuitbreaker/operator/MonoCircuitBreaker.java#L25